src/ray/common/asio/asio_chaos.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/common/asio/asio_chaos.h"
+
+#include <cstdlib>
+#include <string_view>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+#include "ray/common/ray_config.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+namespace asio {
+namespace testing {
+
+namespace {
+/*
+  DelayManager is a simple chaos testing framework. Before starting ray, users
+  should set up os environment to use this feature for testing purposes.
+
+  To use this, simply do
+      export RAY_testing_asio_delay_ms="method1=10,method2=20"
+
+   The delay is a random number between 0 and the value. If method equals '*',
```

Comment:
Any reason why we set this random? I feel like deterministic number is easier to reproduce issues.

Reply:
Maybe support syntax like 0-10 for random, and make it deterministic by default? Wdyt?

Reply:
I guess random for now is also fine since this might require more work

Reply:
Deterministic delays defeat the purpose of randomization here

Reply:
Thinking more about it, I think we can just do range value support (3-5 for example instead of only having 0-5). And in this way we can do 5-5 for deterministic delay (and 5 automatically means 0-5). 

I just think this feature is useful not just for randoimnized chaos testing but also for reliably reproducing certain issues or simulating load (in this case 30-50 microsecond is more realistic than 0-50 microsecond I believe). No need to do it now. I can just add it when I think I want this feature

Reply:
I think the range value here makes sense. Let me update it. So it'll be something like `method1=v1:v2`